lib/queue.rb

```diff
@@ -1,16 +1,47 @@
 class Queue
 
+  MAX_SIZE = 20
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @queue = []
+    @size = 20
+    @front = -1 
+    @rear = -1
+    
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+
+    if (@front == 0 && @rear == @size - 1) 
+    raise "queue is full"
+    elsif (@front == -1)
+      @front = @rear = 0
+      @queue[@rear] = element
+    else
+      @rear += 1
+      @queue[@rear] = element
+    end
+
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if (@front == -1) 
+      raise "queue is empty"
+    end
+    
+    result = @queue[@front]
+    @queue[@front] = nil
+  
+    #if the queue is now empty
+    if (@front == @rear)
+      @front = -1
+      @rear = -1
+    elsif (@front == @size-1) 
+      @front = 0
+    else
+      @front = @front + 1
+    end
```

Comment:
This will let front wrap around without having to make the elif-else.

```suggestion
    else
      @front = (@front + 1) % @store.length
    end
```